File: frontend/src/types/message.ts
export interface UserBasic {
  id: string;
  full_name: string;
  avatar_url?: string | null;
}

export interface ListingBasic {
  id: string;
  vehicle_id: string;
}

export interface Message {
  id: string;
  sender_id: string;
  recipient_id: string;
  listing_id?: string | null;
  content: string;
  is_read: boolean;
  created_at: string;
}

export interface ConversationContact {
  contact: UserBasic;
  listing: ListingBasic | null;
  last_message: Message;
  unread_count: number;
}

export interface MessageCreate {
  recipient_id: string;
  listing_id?: string;
  content: string;
}

File: backend/app/models/message.py
from sqlalchemy import Column, String, Text, Boolean, ForeignKey, DateTime
from sqlalchemy.orm import relationship
import uuid
from datetime import datetime, timezone
from app.core.database import Base

class Message(Base):
    __tablename__ = "messages"

    id = Column(String, primary_key=True, default=lambda: str(uuid.uuid4()))
    sender_id = Column(String, ForeignKey("users.id"), nullable=False)
    recipient_id = Column(String, ForeignKey("users.id"), nullable=False)
    listing_id = Column(String, ForeignKey("listings.id"), nullable=True) # Optional, provides context
    content = Column(Text, nullable=False)
    is_read = Column(Boolean, default=False)
    created_at = Column(DateTime(timezone=True), default=lambda: datetime.now(timezone.utc))

    # Relationships
    sender = relationship("User", foreign_keys=[sender_id])
    recipient = relationship("User", foreign_keys=[recipient_id])
    listing = relationship("Listing", foreign_keys=[listing_id])
